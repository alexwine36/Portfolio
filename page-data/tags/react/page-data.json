{
    "componentChunkName": "component---src-components-pages-tags-index-tsx",
    "path": "/tags/react/",
    "result": {"data":{"projects":{"totalCount":5,"nodes":[{"frontmatter":{"tags":["Pulumi","AWS","Step Functions","React","Node JS"],"title":"Customer Surveys","hero":{"childImageSharp":{"gatsbyImageData":{"layout":"constrained","placeholder":{"fallback":"data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAANABQDASIAAhEBAxEB/8QAGQAAAgMBAAAAAAAAAAAAAAAAAAQBAgMF/8QAFgEBAQEAAAAAAAAAAAAAAAAAAQAC/9oADAMBAAIQAxAAAAHoVUWcwYk//8QAHBAAAgEFAQAAAAAAAAAAAAAAAgMBAAQQERIy/9oACAEBAAEFApuRiNdixxLZ5osf/8QAFBEBAAAAAAAAAAAAAAAAAAAAEP/aAAgBAwEBPwE//8QAFBEBAAAAAAAAAAAAAAAAAAAAEP/aAAgBAgEBPwE//8QAHBAAAQMFAAAAAAAAAAAAAAAAAAECEhARITFB/9oACAEBAAY/AiUrDmt0i9MV/8QAGhAAAwEBAQEAAAAAAAAAAAAAAAERIaExUf/aAAgBAQABPyH6wpdK2IS8ng02YkWdN+8SREf/2gAMAwEAAgADAAAAEF/P/8QAFREBAQAAAAAAAAAAAAAAAAAAABH/2gAIAQMBAT8Qqv/EABURAQEAAAAAAAAAAAAAAAAAAAAR/9oACAECAQE/EIj/xAAeEAEAAgEEAwAAAAAAAAAAAAABABEhMUFhkVGhwf/aAAgBAQABPxDCNupdEuunMHA81ekeAwR7VyroHL9DArIrx7AI2YHuf//Z"},"images":{"fallback":{"src":"/Portfolio/static/793b780233247c986b7188e24e61f0c0/5a755/lukas-blazek-mcsdtbwxuzu-unsplash.jpg","srcSet":"/Portfolio/static/793b780233247c986b7188e24e61f0c0/81053/lukas-blazek-mcsdtbwxuzu-unsplash.jpg 1536w,\n/Portfolio/static/793b780233247c986b7188e24e61f0c0/835ac/lukas-blazek-mcsdtbwxuzu-unsplash.jpg 3072w,\n/Portfolio/static/793b780233247c986b7188e24e61f0c0/5a755/lukas-blazek-mcsdtbwxuzu-unsplash.jpg 6144w","sizes":"(min-width: 6144px) 6144px, 100vw"},"sources":[{"srcSet":"/Portfolio/static/793b780233247c986b7188e24e61f0c0/7fa70/lukas-blazek-mcsdtbwxuzu-unsplash.webp 1536w,\n/Portfolio/static/793b780233247c986b7188e24e61f0c0/b0586/lukas-blazek-mcsdtbwxuzu-unsplash.webp 3072w,\n/Portfolio/static/793b780233247c986b7188e24e61f0c0/5366d/lukas-blazek-mcsdtbwxuzu-unsplash.webp 6144w","type":"image/webp","sizes":"(min-width: 6144px) 6144px, 100vw"}]},"width":6144,"height":4068.9999999999995}}}},"excerpt":"The Problem Construction is a long process and making sure a customer is happy throughout a build is challenging and the highest priority…","slug":"customer-surveys/","timeToRead":1},{"frontmatter":{"tags":["Serverless","Typescript","Node JS","React"],"title":"Receipt Manager","hero":{"childImageSharp":{"gatsbyImageData":{"layout":"constrained","placeholder":{"fallback":"data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAANABQDASIAAhEBAxEB/8QAGAAAAgMAAAAAAAAAAAAAAAAAAAIBAwX/xAAVAQEBAAAAAAAAAAAAAAAAAAABAv/aAAwDAQACEAMQAAABVyxnPIE//8QAGBAAAwEBAAAAAAAAAAAAAAAAAQIDABH/2gAIAQEAAQUClJghiwzcBhR9W7jG1O//xAAUEQEAAAAAAAAAAAAAAAAAAAAQ/9oACAEDAQE/AT//xAAUEQEAAAAAAAAAAAAAAAAAAAAQ/9oACAECAQE/AT//xAAaEAACAgMAAAAAAAAAAAAAAAAAMSEyERKR/9oACAEBAAY/AqsRO3B5ILH/xAAdEAEAAgEFAQAAAAAAAAAAAAABABExIWFxkaHB/9oACAEBAAE/IXl/UgwWPYHgtvGMvlNLR1GZJ//aAAwDAQACAAMAAAAQ78//xAAWEQEBAQAAAAAAAAAAAAAAAAARAAH/2gAIAQMBAT8QwJv/xAAXEQEAAwAAAAAAAAAAAAAAAAAAESFB/9oACAECAQE/ELxL/8QAHBABAAMAAgMAAAAAAAAAAAAAAQARITFBcYHB/9oACAEBAAE/EFB64IZeOoPNm0tvkqaTuTBWArh3EQu+6xQvron/2Q=="},"images":{"fallback":{"src":"/Portfolio/static/273021880b58d85211a7f2e692a6095c/06b4d/scott-graham-oqmzwnd3thu-unsplash.jpg","srcSet":"/Portfolio/static/273021880b58d85211a7f2e692a6095c/e2514/scott-graham-oqmzwnd3thu-unsplash.jpg 1840w,\n/Portfolio/static/273021880b58d85211a7f2e692a6095c/89a98/scott-graham-oqmzwnd3thu-unsplash.jpg 3680w,\n/Portfolio/static/273021880b58d85211a7f2e692a6095c/06b4d/scott-graham-oqmzwnd3thu-unsplash.jpg 7360w","sizes":"(min-width: 7360px) 7360px, 100vw"},"sources":[{"srcSet":"/Portfolio/static/273021880b58d85211a7f2e692a6095c/4dbab/scott-graham-oqmzwnd3thu-unsplash.webp 1840w,\n/Portfolio/static/273021880b58d85211a7f2e692a6095c/f5714/scott-graham-oqmzwnd3thu-unsplash.webp 3680w,\n/Portfolio/static/273021880b58d85211a7f2e692a6095c/04abc/scott-graham-oqmzwnd3thu-unsplash.webp 7360w","type":"image/webp","sizes":"(min-width: 7360px) 7360px, 100vw"}]},"width":7360,"height":4912}}}},"excerpt":"The Problem Bills from utilities are all mailed Need to be coded to houses The Solution Scanned and emailed to account on AWS Stored in AWS…","slug":"receipt-manager/","timeToRead":1},{"frontmatter":{"tags":["React","Typescript","Prisma","GraphQL"],"title":"Start Pack Generator","hero":{"childImageSharp":{"gatsbyImageData":{"layout":"constrained","placeholder":{"fallback":"data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAeABQDASIAAhEBAxEB/8QAGQAAAgMBAAAAAAAAAAAAAAAAAAQBAgUD/8QAFwEBAQEBAAAAAAAAAAAAAAAAAAIBA//aAAwDAQACEAMQAAABemMy2gIhfltU50mMk7//xAAZEAACAwEAAAAAAAAAAAAAAAAAAQIDERL/2gAIAQEAAQUCwwwc1E5n1t5bRhGNqEpZo4pnKP/EABcRAQEBAQAAAAAAAAAAAAAAAAARASH/2gAIAQMBAT8BiMjj/8QAGREAAQUAAAAAAAAAAAAAAAAAAAIQESJB/9oACAECAQE/AWVOFz//xAAjEAAAAwYHAAAAAAAAAAAAAAAAAQIQERIhMZEDIjJRcYGx/9oACAEBAAY/Am1J+wifiF2NarEI1KmMvgmU+GVVcVVcf//EAB8QAAMAAgICAwAAAAAAAAAAAAABESFBMWEQUXGBwf/aAAgBAQABPyFBh5CJx8bZOEd7Qlt+wWjo+Bwhx7z/AElZdmI7GlhfXgf/2gAMAwEAAgADAAAAEDQd8P/EABkRAAIDAQAAAAAAAAAAAAAAAAABEBExUf/aAAgBAwEBPxCBk0vk/8QAGREAAgMBAAAAAAAAAAAAAAAAABEBEGGh/9oACAECAQE/EKbBWen/xAAgEAEAAgEEAgMAAAAAAAAAAAABABEhMUFRkXHBYYGh/9oACAEBAAE/ECqYdHqDaB4uxVfwRRWepj54tIsoJy/uuLoFMuBnZ9fEprJCMl+4CT1aEt6mC7HiMN8VUdRt9qf/2Q=="},"images":{"fallback":{"src":"/Portfolio/static/9529386388c8a16b61e9886bb64433ff/db4ed/sharon-mccutcheon-tn57JI3CewI-unsplash.jpg","srcSet":"/Portfolio/static/9529386388c8a16b61e9886bb64433ff/73214/sharon-mccutcheon-tn57JI3CewI-unsplash.jpg 936w,\n/Portfolio/static/9529386388c8a16b61e9886bb64433ff/5f79b/sharon-mccutcheon-tn57JI3CewI-unsplash.jpg 1872w,\n/Portfolio/static/9529386388c8a16b61e9886bb64433ff/db4ed/sharon-mccutcheon-tn57JI3CewI-unsplash.jpg 3744w","sizes":"(min-width: 3744px) 3744px, 100vw"},"sources":[{"srcSet":"/Portfolio/static/9529386388c8a16b61e9886bb64433ff/7291c/sharon-mccutcheon-tn57JI3CewI-unsplash.webp 936w,\n/Portfolio/static/9529386388c8a16b61e9886bb64433ff/76833/sharon-mccutcheon-tn57JI3CewI-unsplash.webp 1872w,\n/Portfolio/static/9529386388c8a16b61e9886bb64433ff/b491a/sharon-mccutcheon-tn57JI3CewI-unsplash.webp 3744w","type":"image/webp","sizes":"(min-width: 3744px) 3744px, 100vw"}]},"width":3744,"height":5616}}}},"excerpt":"The Problem Starpacks used for job starts Used for preconstruction meeting The Solution Includes all options for a house Includes colors for…","slug":"start-pack-generator/","timeToRead":1},{"frontmatter":{"tags":["React","Typescript","Tableau"],"title":"Tableau Extensions","hero":{"childImageSharp":{"gatsbyImageData":{"layout":"constrained","placeholder":{"fallback":"data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAOABQDASIAAhEBAxEB/8QAFwABAQEBAAAAAAAAAAAAAAAAAgAEBf/EABUBAQEAAAAAAAAAAAAAAAAAAAAB/9oADAMBAAIQAxAAAAElA13RrP/EABoQAAICAwAAAAAAAAAAAAAAAAECAyIAERL/2gAIAQEAAQUCXaL3Jixs4AJVReMU/8QAFBEBAAAAAAAAAAAAAAAAAAAAEP/aAAgBAwEBPwE//8QAFREBAQAAAAAAAAAAAAAAAAAAABH/2gAIAQIBAT8Bqv/EABsQAAICAwEAAAAAAAAAAAAAAAABAhEhQWGR/9oACAEBAAY/AqUTfhllxYoi6f/EABoQAAMBAQEBAAAAAAAAAAAAAAABESFRQXH/2gAIAQEAAT8hj0dY54kvAktlWOjTg4yvgqRtfR//2gAMAwEAAgADAAAAEKMv/8QAFhEBAQEAAAAAAAAAAAAAAAAAAQAR/9oACAEDAQE/EEsL/8QAFhEBAQEAAAAAAAAAAAAAAAAAAQAh/9oACAECAQE/ECLML//EAB4QAQACAgIDAQAAAAAAAAAAAAEAESExQZFRcYGh/9oACAEBAAE/EBiU4Vt5j9wNFVHUdnMZZurf1qXQWaa9+YjztM9wDlBf1uf/2Q=="},"images":{"fallback":{"src":"/Portfolio/static/a2a7d1b8895ebb809f2767040f31cd62/c5e1e/stephen-dawson-qwtCeJ5cLYs-unsplash.jpg","srcSet":"/Portfolio/static/a2a7d1b8895ebb809f2767040f31cd62/df3c1/stephen-dawson-qwtCeJ5cLYs-unsplash.jpg 737w,\n/Portfolio/static/a2a7d1b8895ebb809f2767040f31cd62/100ce/stephen-dawson-qwtCeJ5cLYs-unsplash.jpg 1474w,\n/Portfolio/static/a2a7d1b8895ebb809f2767040f31cd62/c5e1e/stephen-dawson-qwtCeJ5cLYs-unsplash.jpg 2947w","sizes":"(min-width: 2947px) 2947px, 100vw"},"sources":[{"srcSet":"/Portfolio/static/a2a7d1b8895ebb809f2767040f31cd62/e4d84/stephen-dawson-qwtCeJ5cLYs-unsplash.webp 737w,\n/Portfolio/static/a2a7d1b8895ebb809f2767040f31cd62/75990/stephen-dawson-qwtCeJ5cLYs-unsplash.webp 1474w,\n/Portfolio/static/a2a7d1b8895ebb809f2767040f31cd62/dc15f/stephen-dawson-qwtCeJ5cLYs-unsplash.webp 2947w","type":"image/webp","sizes":"(min-width: 2947px) 2947px, 100vw"}]},"width":2947,"height":2121}}}},"excerpt":"The Problem Tableau doesn't always have the necessary flexibilty to display data","slug":"tableau-extensions/","timeToRead":1},{"frontmatter":{"tags":["React","Typescript","GraphQL","PostgreSQL"],"title":"Temp Check","hero":null},"excerpt":"The problem local school needed a way to log children’s temperature quickly during drop off the solution prisma orm ","slug":"temp-check/","timeToRead":1}]},"skill":{"date":"9 years ago","rating":5,"skill":"React"}},"pageContext":{"tag":"React"}},
    "staticQueryHashes": ["374626863"]}