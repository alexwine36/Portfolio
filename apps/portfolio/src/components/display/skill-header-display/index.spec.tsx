import { render } from '@testing-library/react';
import React from 'react';
import SkillHeaderDisplay from './index';

describe('SkillHeaderDisplay', () => {
  it('should render successfully', () => {
    const { baseElement } = render(
      <SkillHeaderDisplay
        skill={{
          date: '1 year ago',
          rating: 3,
          skill: 'Sample',
        }}
      />
    );
    expect(baseElement).toBeTruthy();
  });
  // test('should format include correct data', () => {
  //   const { getByTestId } = render(
  //     <SkillHeaderDisplay
  //       skill={{
  //         date: '1 year ago',
  //         rating: 3,
  //         skill: 'Sample',
  //       }}
  //     />
  //   );
  //   const header = getByTestId('skill');
  //   expect(header.textContent).toBe('Sample');

  //   const time = getByTestId('skill-time');
  //   expect(time.textContent).toBe('~1 year');
  //   // console.log(container.firstChild.firstChild);
  //   const holder = getByTestId('container');
  //   console.log(holder.innerHTML);

  //   expect(holder).toMatchSnapshot();
  //   // const label = within(holder).getByLabelText('3 Stars');
  //   // const rating = getByTestId('rating');
  //   // // console.log(rating);
  //   // const label = within(rating);
  //   // expect(label).toBeTruthy();
  // });
});
