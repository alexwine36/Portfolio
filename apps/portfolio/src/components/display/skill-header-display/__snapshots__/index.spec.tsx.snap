// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkillHeaderDisplay should format include correct data 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  font-size: 1.5rem;
  color: #faaf00;
  cursor: pointer;
  text-align: left;
  -webkit-tap-highlight-color: transparent;
  pointer-events: none;
}

.emotion-1.Mui-disabled {
  opacity: 0.38;
  pointer-events: none;
}

.emotion-1.Mui-focusVisible .MuiRating-iconActive {
  outline: 1px solid #999;
}

.emotion-1 .MuiRating-visuallyHidden {
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-transition: -webkit-transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  pointer-events: none;
}

.emotion-3 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: inherit;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-transition: -webkit-transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  pointer-events: none;
  color: rgba(0, 0, 0, 0.26);
}

.emotion-12 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
}

<div
  class="MuiBox-root emotion-0"
  data-testid="container"
>
  <span
    aria-label="3 Stars"
    class="MuiRating-root MuiRating-sizeMedium MuiRating-readyOnly emotion-1"
    data-testid="rating"
    role="img"
  >
    <span>
      <span
        class="MuiRating-icon MuiRating-iconFilled emotion-2"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit emotion-3"
          data-testid="StarIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
      </span>
    </span>
    <span>
      <span
        class="MuiRating-icon MuiRating-iconFilled emotion-2"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit emotion-3"
          data-testid="StarIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
      </span>
    </span>
    <span>
      <span
        class="MuiRating-icon MuiRating-iconFilled emotion-2"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit emotion-3"
          data-testid="StarIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
      </span>
    </span>
    <span>
      <span
        class="MuiRating-icon MuiRating-iconEmpty emotion-8"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit emotion-3"
          data-testid="StarBorderIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
          />
        </svg>
      </span>
    </span>
    <span>
      <span
        class="MuiRating-icon MuiRating-iconEmpty emotion-8"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit emotion-3"
          data-testid="StarBorderIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
          />
        </svg>
      </span>
    </span>
  </span>
  <p
    class="MuiTypography-root MuiTypography-body1 emotion-12"
    data-testid="skill"
  >
    Sample
  </p>
</div>
`;
